packages/runtime/src/App.tsx

```diff
@@ -135,9 +144,11 @@ export const ImplWrapper = React.forwardRef<
 
   // eval component properties
   useEffect(() => {
-    const { stop } = deepEval(c.properties, ({ result }) => {
-      setEvaledComponentProperties({ ...result });
+    const { result, stop } = deepEval(c.properties, ({ result: newResult }) => {
+      setEvaledComponentProperties({ ...newResult });
     });
+    // must keep this line, reason is the same as above
+    setEvaledComponentProperties({ ...result });
```

Comment:
Good catch, also run into this issue in my branch.

Another issue that may not need to be changed in this PR is shall we remove the `deepEval` call to get the initial value of `evaledComponentProperties`?

Reply:
never mind, just found it has already been removed.

Reply:
update: it should be kept to make sure the component gets evaluated property at the first render.